mobile_verifier/src/reward_shares.rs

```diff
@@ -220,19 +236,63 @@ impl MapperShares {
         reward_period: &'_ Range<DateTime<Utc>>,
         reward_per_share: Decimal,
     ) -> impl Iterator<Item = (u64, proto::MobileRewardShare)> + '_ {
-        self.discovery_mapping_shares
+        let mut subscriber_rewards: HashMap<Vec<u8>, proto::SubscriberReward> = HashMap::new();
+
+        // Accumulate rewards from discovery_mapping_shares
+        for subscriber_id in self.discovery_mapping_shares {
+            let discovery_location_amount = (DISCOVERY_MAPPING_SHARES * reward_per_share)
+                .round_dp_with_strategy(0, RoundingStrategy::ToZero)
+                .to_u64()
+                .unwrap_or_default();
+
+            if discovery_location_amount > 0 {
+                subscriber_rewards
+                    .entry(subscriber_id.clone())
+                    .and_modify(|reward| {
+                        reward.discovery_location_amount += discovery_location_amount;
+                    })
+                    .or_insert_with(|| proto::SubscriberReward {
+                        subscriber_id: subscriber_id.clone(),
+                        discovery_location_amount,
+                        verification_mapping_amount: 0,
+                    });
+            }
+        }
+
+        // Accumulate rewards from verified_mapping_event_shares
+        for verified_share in self.verified_mapping_event_shares {
+            let verification_mapping_amount = (Decimal::from(verified_share.total_reward_points)
+                * reward_per_share)
+                .round_dp_with_strategy(0, RoundingStrategy::ToZero)
+                .to_u64()
+                .unwrap_or_default();
+
+            if verification_mapping_amount > 0 {
+                subscriber_rewards
+                    .entry(verified_share.subscriber_id.clone())
+                    .and_modify(|reward| {
+                        reward.verification_mapping_amount += verification_mapping_amount;
```

Comment:
Should this replace the default value rather than adding to it ? I get that the default value will be zero but logically it doesnt seem right to add instead of replace

Other than that this latest change looks good

Reply:
Idk to be honest what should be the expected behavior at this point. I guess it depends if we can receive more than one but I think I remember that normally it's not possible.

Reply:
verified_mapping_event_shares will contain only a single entry per subscriber as per https://github.com/helium/oracles/blob/63dad9aa2d02cdf74220663e53e38f95482611fa/mobile_verifier/src/verified_subscriber_mapping_event.rs#L126

So whilst the ingestor should only receive a single verified mapping event per epoch, should for whatever reason more than one be received they are aggregated per subscriber in the above into a single entry